src/main/cpp/syslogappender.cpp

```diff
@@ -326,11 +327,12 @@ void SyslogAppender::append(const spi::LoggingEventPtr& event, Pool& p)
 
 		while ( start != msg.end() )
 		{
-			LogString::iterator end = start + maxMessageLength - 12;
+			LogString::iterator end;
 
-			if ( end > msg.end() )
-			{
+			if( msg.size() > (maxMessageLength - MINIMUM_MESSAGE_SIZE) ) {
```

Comment:
Line 349 contains the following line which might be changed to using `MINIMUM_MESSAGE_SIZE` instead:

    char buf[12];